### DISCUSSION TITLE
Conditional Routes / Filtering Routes / Requires Page Reload?

### DISCUSSION THREAD

**Discussion by alecabaker** [tighten/ziggy#463] [Q&A]
#### Description

In a Laravel blade file, we conditionally load our routes.  See below for the Ziggy call and context.

Why? Well, if the user is not authenticated we only want them to have access to the authentication routes and not be able to view the page source and see all routes.  This works as expected with our VueJS/InertiaJS application, however, once the user authenticates the user can't access the new routes they should now have access to without refreshing the page.  This refresh then updates the route list with the new routes they have access to.

```
@if(auth()->user())
   @routes
@else
   @routes('guest')
@endif
```

For what it's worth, the example above is illustratory only.  We have multiple user types/groups.

Is there a better way to implement Ziggy to avoid the page reload while better protecting routes?

Thank you in advance!

**Comment by bakerkretzmar** [comment]
Hey, yeah in this case you're probably better off not using the `@routes` Blade directive and instead passing the routes along with the Inertia response. Try including it in your Inertia [shared data](https://inertiajs.com/shared-data):

```php
use Tightenco\Ziggy\Ziggy;

class HandleInertiaRequests extends Middleware
{
    public function share(Request $request)
    {
        return array_merge(parent::share($request), [
            'ziggy' => fn () => (new Ziggy($request->user() ? null : 'guest'))->toArray(),
        ]);
    }
}
```

You may also need to set up a custom Vue mixin to make sure that your JS `route()` function is always using the latest Ziggy config from Inertia.

Does that make sense? Let me know if you get that working, if you're stuck I can try to set up a demo of how this could work (I didn't test the code above I'm just thinking out loud 😅).

Also, you'll definitely want to cache the generated routes, probably per-group, because generating them can be expensive. With `@routes` you get this for free since the entire output is dumped into the view, which is then cached.

**Reply by faso-dev** [reply]
Hello, i still facing the same issue with intertia react js.

laravel 11 and latest inertia js.

```"@inertiajs/react": "^2.0.5"```

so how can make it work as when i reload the page, i see all routes, : 

in ziggy config   : 

```php
<?php

// config/ziggy.php

return [
    'groups' => [
        'admin' => ['admin.*'],
        'guest' => ['website.*'],
        'user' => ['user.*'],
    ],
];
```

in blade : 

```php
// app.blade.php

     @if (is_admin())
            @routes(['admin'])
     @elseif (is_user())
            @routes(['user', 'website'])
     @else
            @routes(['website'])
      @endif
 
 ```
 
 so when i inspect the page source, i see all both routes (admin.* and website.*), ```is_admin``` juste check if user is admin.

**Reply by bakerkretzmar** [reply]
Same thing as my earlier answer, you should probably not use `@routes` in your case. However if you're seeing all your routes on every page, your `is_admin()` or `is_user()` functions probably aren't working exactly how you want.

Also, you don't have a route group named `website`, so maybe you need to use `guest` instead in your `app.blade.php` or update your config.

I would strongly suggest just using `@routes` with no groups and loading all your routes on every page though if possible, it's much simpler.
